internal/lm/nvml.go

```diff
@@ -218,6 +224,45 @@ func newGPUModeLabeler(devices []resource.Device) (Labeler, error) {
 	return labels, nil
 }
 
+func newImexDomainLabeler(devices []resource.Device) (Labeler, error) {
+	deviceTuple := map[string]string{
+		"clusterUuid": "",
+		"cliqueId":    "",
+	}
+	for _, d := range devices {
+		clusterUuid, err := d.GetClusterUUID()
+		if err != nil {
+			return nil, err
+		}
+		if deviceTuple["clusterUuid"] == "" {
+			deviceTuple["clusterUuid"] = clusterUuid
+		} else if deviceTuple["clusterUuid"] != clusterUuid {
+			klog.Warningf("cluster UUID mismatch: %s != %s", deviceTuple["clusterUuid"], clusterUuid)
+			return nil, fmt.Errorf("cluster UUID mismatch: %s != %s", deviceTuple["clusterUuid"], clusterUuid)
+		}
```

Comment:
Can you drop the else here and just do an if -- the else makes it harder to read. I'm not worried about the fact that one time around the loop we do an extra if check. Same below.

Reply:
done